PVB Linx Groep 6/Assets/Code/Scripts/UI/Shop/InteractionShop.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using UnityEngine;
+
+namespace Code.Scripts.UI.Shop
+{
+    public class InteractionShop : MonoBehaviour
+    {
+        
+        [SerializeField] private GameObject shopUI;
+        public GameObject currentBuyer;
+        private bool _inRange;
+
+        private void Awake() => shopUI.SetActive(false);
+        
+        private void Update()
+        {
+            if (Input.GetKeyDown(KeyCode.E) && _inRange)
+                shopUI.SetActive(true);
+        }
+        
+        private void OnTriggerEnter(Collider other)
+        {
+            if (other.CompareTag("Player"))
+            {
+                currentBuyer = other.gameObject;
```

Comment:
this cant be assinged here as this could cause currentbuyer to be a different player from the one opening the shop